### PR TITLE
[3/n][dagster-fivetran] Implement FivetranClient for rework

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR implements `FivetranClient`, which is based on the legacy `FivetranResource` code. Basically reusing how `make_request` was implemented and adding more methods to leverage in a subsequent PR to fetch the Fivetran workspace data.

## How I Tested These Changes

Additional unit test

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25756?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25756** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25756?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25751** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25751?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25750** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25750?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by benpankow — APPROVED** [review]
Nice, looks good to me

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+@pytest.fixture(name="group_id")
+def group_id_fixture() -> str:
+    return "group_id"
+
+
+@pytest.fixture(
+    name="workspace_data_api_mocks_fn",
+)
+def workspace_data_api_mocks_fn_fixture(
```
Comment (dpeng817) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py:224]: In general I find API mocks like this to be relatively inflexible and hard to parse. How would you feel about instead implementing a "client fake" similar to how we did for airlift and dlift? Something like this:
https://github.com/dagster-io/dagster/blob/master/examples/experimental/dagster-dlift/dagster_dlift/test/client_fake.py

I've found it to be more composable and human-readable than mocking. I won't necessarily block review on this, but every time I've based a testing strategy around mocks I've regretted it. I assume this is also somewhat inherited from the old testing strategy, but I think if we're spending time in this code we might as well improve the situation.

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+            "port": 0,
+            "databricks_connection_type": "Directly | PrivateLink | SshTunnel | ProxyAgent",
+            "secret_value": "service_principal_secret_value",
+            "oauth2_client_id": "string",
+            "personal_access_token": "string",
+        },
+    },
+}
+
+SAMPLE_CONNECTOR_DETAILS = {
```
Comment (dpeng817) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py:140]: comment on how these were generated? What is our plan if the schema changes, etc?

**Review by dpeng817 — CHANGES_REQUESTED** [review]
I won't block review on this, but I'd like to get your thoughts regarding testing strategy here. 

One thing that we also started to do with dlift + airlift was have some test battery running live against the integration. IMO when a new person onboards to the integration, something like that is pretty invaluable in terms of getting started with the tooling, I will always trust it better than just pure unit tests especially when running against an external system which can change at any point. The _lack_of these I think is a major reason why we don't feel confidence in integrations which haven't been touched in a while. Doesn't have to be super complicated, but having a "kitchen sink" testbed for the integration similar to what we did for dlift/airlift: https://github.com/dagster-io/dagster/blob/master/examples/experimental/dagster-dlift/kitchen-sink/Makefile could be very nice.

Thoughts?

**Review by dpeng817 — APPROVED** [review]
Oops meant to approve.

**Comment by maximearmstrong** [comment]
> One thing that we also started to do with dlift + airlift was have some test battery running live against the integration. IMO when a new person onboards to the integration, something like that is pretty invaluable in terms of getting started with the tooling, I will always trust it better than just pure unit tests especially when running against an external system which can change at any point. The _lack_of these I think is a major reason why we don't feel confidence in integrations which haven't been touched in a while. Doesn't have to be super complicated, but having a "kitchen sink" testbed for the integration similar to what we did for dlift/airlift: https://github.com/dagster-io/dagster/blob/master/examples/experimental/dagster-dlift/kitchen-sink/Makefile could be very nice. Then, you can leverage `dagster-buildkite` to make it only run in the nightly + on PRs that touch integration code.
> 
> Thoughts?

I'm very much in favor of this strategy. I agree that unit tests mocking external services are not enough to make sure that the integration works properly. I'll open a ticket for that.

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py (hunk 2 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py:147]: Updated in [e4892e8](https://github.com/dagster-io/dagster/pull/25756/commits/e4892e84bdd3bc520af23b573d2e92b6f6fab320)

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py (hunk 1 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py]: See [comment](https://github.com/dagster-io/dagster/pull/25788#discussion_r1835003439) in next PR
